File: old_groovy/src/main/groovy/net/zomis/brainf/model/ast/tree/ReadSyntax.java
package net.zomis.brainf.model.ast.tree;

import net.zomis.brainf.model.BrainfuckException;
import net.zomis.brainf.model.BrainfuckRunner;

public class ReadSyntax extends Syntax {

    @Override
    public void perform(BrainfuckRunner runner) {
        int value;
        try {
            value = runner.getInput().read();
        } catch (BrainfuckException e) {
            throw new RuntimeException(e);
        }
        runner.getMemory().setMemory(value);
    }

}

File: brainduck-kotlin/src/commonMain/kotlin/net/zomis/brainf/model/ast/tree/ReadSyntax.kt
package net.zomis.brainf.model.ast.tree;

import net.zomis.brainf.model.BrainfuckRunner;

class ReadSyntax : Syntax() {

    override fun perform(runner: BrainfuckRunner) {
        val value = runner.input.read()
        runner.memory.setMemory(value)
    }

}
